src/Aspire.Hosting/LaunchProfileExtensions.cs

```diff
@@ -25,34 +25,45 @@ internal static class LaunchProfileExtensions
             return null;
         }
 
-        return projectMetadata.GetLaunchSettings();
+        return projectMetadata.GetLaunchSettings(projectResource.Name);
     }
 
     internal static NamedLaunchProfile? GetEffectiveLaunchProfile(this ProjectResource projectResource, bool throwIfNotFound = false)
     {
-        string? launchProfileName = projectResource.SelectLaunchProfileName();
-        if (string.IsNullOrEmpty(launchProfileName))
+        try
         {
-            return null;
-        }
+            string? launchProfileName = projectResource.SelectLaunchProfileName();
+            if (string.IsNullOrEmpty(launchProfileName))
+            {
+                return null;
+            }
 
-        var profiles = projectResource.GetLaunchSettings()?.Profiles;
-        if (profiles is null)
-        {
-            return null;
-        }
+            var profiles = projectResource.GetLaunchSettings()?.Profiles;
+            if (profiles is null)
+            {
+                return null;
+            }
 
-        var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
-        if (!found && throwIfNotFound)
+            var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
+            if (!found && throwIfNotFound)
+            {
+                var message = string.Format(CultureInfo.InvariantCulture, Resources.LaunchSettingsFileDoesNotContainProfileExceptionMessage, launchProfileName);
+                throw new DistributedApplicationException(message);
+            }
+
+            return launchProfile is not null ? new (launchProfileName, launchProfile) : default;
+        }
+        catch (JsonException ex)
```

Comment:
Do we still need this try-catch here now that we are catching the JsonException around the call to `JsonSerializer.Deserialize`?

Reply:
Good catch ... I forgot to trim that code out!